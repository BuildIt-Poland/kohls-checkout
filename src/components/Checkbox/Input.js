import styled from 'styled-components';

import { spacingSmall } from '../../styles/designTokens';

const Input = styled.input`
  visibility: hidden;
  width: 0;
  margin-right: ${spacingSmall};
`;

export default Input;
