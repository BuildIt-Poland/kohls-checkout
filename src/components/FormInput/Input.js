import styled from 'styled-components';

import InputBase from '../Input';

const Input = styled(InputBase)`
  margin-top: 0.4rem;
`;

export default Input;
