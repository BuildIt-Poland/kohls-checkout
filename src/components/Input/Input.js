import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  spacingSmall,
  borderWidthThin,
  borderRadiusLarge,
  colorDarkGrey,
  colorBlack,
  colorRed
} from '../../styles/designTokens';

const StyledInput = styled.input`
  border: ${borderWidthThin} solid ${props => (props.error ? colorRed : colorBlack)};
  border-radius: ${borderRadiusLarge};
  padding: ${spacingSmall};
  outline: 0;
  width: 100%;

  :disabled {
    background-color: ${colorDarkGrey};
  }
`;

const Label = styled.label`
  margin-bottom: ${spacingSmall};
`;

const ErrorLabel = styled.div`
  font-style: italic;
  color: ${colorRed};
`;

const Input = ({ className, label, error, ...props }) => {
  return (
    <>
      {label ? (
        <Label className={className}>
          {label}
          <StyledInput error={error} {...props} />
        </Label>
      ) : (
        <StyledInput error={error} {...props} />
      )}
      {error && <ErrorLabel>{error}</ErrorLabel>}
    </>
  );
};

Input.propTypes = {
  label: PropTypes.string,
  error: PropTypes.string,
  type: PropTypes.string.isRequired,
  className: PropTypes.any
};

export default Input;
