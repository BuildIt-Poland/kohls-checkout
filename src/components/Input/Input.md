```js
<>
  <Input placeholder="Basic input" />
  <Input placeholder="Errored input" error />
  <Input placeholder="Disabled input" disabled />
</>
```
