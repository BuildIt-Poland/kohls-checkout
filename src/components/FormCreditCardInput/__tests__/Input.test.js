import React from 'react';
import { create } from 'react-test-renderer';

import Input from '../Input';

describe('COMPONENT - FormCreditCardInput Input', () => {
  it('renders correctly', () => {
    const component = create(<Input />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
