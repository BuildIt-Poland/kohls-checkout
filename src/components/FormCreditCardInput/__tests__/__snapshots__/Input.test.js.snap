// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - FormCreditCardInput Input renders correctly 1`] = `
.c0 {
  border: 1px solid #b0b0b0;
  border-radius: 0;
  height: 4rem;
  padding: 0 1rem;
  font-size: 1.8rem;
  outline: 0;
  width: 100%;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.25);
  margin-top: 0.4rem;
}

.c0:disabled {
  background-color: #b0b0b0;
}

.c0:focus {
  border-color: #00a09f;
  outline: 4px solid #ffbf47;
  outline-offset: 0;
}

.c0::-webkit-input-placeholder {
  color: #666666;
}

.c0::-moz-placeholder {
  color: #666666;
}

.c0:-ms-input-placeholder {
  color: #666666;
}

.c0::placeholder {
  color: #666666;
}

<input
  className="c0"
/>
`;
